src/iotjs_env.c

```diff
@@ -193,26 +165,24 @@ const Config* iotjs_environment_config(const iotjs_environment_t* env) {
 }
 
 
-void iotjs_environment_go_state_running_main(iotjs_environment_t* env) {
-  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_environment_t, env);
-
-  IOTJS_ASSERT(_this->state == kInitializing);
-  _this->state = kRunningMain;
-}
-
-
-void iotjs_environment_go_state_running_loop(iotjs_environment_t* env) {
-  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_environment_t, env);
-
-  IOTJS_ASSERT(_this->state == kRunningMain);
-  _this->state = kRunningLoop;
-}
-
-
-void iotjs_environment_go_state_exiting(iotjs_environment_t* env) {
+void iotjs_environment_set_state(iotjs_environment_t* env, State s) {
   IOTJS_VALIDATED_STRUCT_METHOD(iotjs_environment_t, env);
-  IOTJS_ASSERT(_this->state < kExiting);
-  _this->state = kExiting;
+  switch (s) {
+    case kInitializing:
```

Comment:
Can you switch to initializing from any state?

Reply:
@zherczeg Debugger make the state to `kInitializing` on resetting context. But it changes the state by assigning `kInitializing` directly.

Reply:
No thank you.